src/api/discounts.js

```diff
@@ -0,0 +1,45 @@
+export const validateDiscount = async (csrf, apiPath, code) => {
+  const response = await fetch(`${apiPath}/validate_discount_code?discount_code=${code}`, {
+    mode: 'cors',
+    method: 'GET',
+    credentials: 'include',
+    headers: {
+      'Content-Type': 'application/json',
+      'X-CSRF-TOKEN': csrf,
+    },
+  });
+
+  return response.json();
+};
+
+export const addDiscount = async (csrf, apiPath, code) => {
+  const response = await fetch(`${apiPath}/discounts`, {
+    mode: 'cors',
+    method: 'POST',
+    credentials: 'include',
+    headers: {
+      'Content-Type': 'application/json',
+      'X-CSRF-TOKEN': csrf,
+    },
+    body: JSON.stringify({
+      code,
+    }),
+  });
+  return response.json();
```

Comment:
I wonder if in the future we should check the status of the response to ensure that there wasn't an error. For instance I know that we have gotten that "invalid '<' token at index 0" error a lot because there was an error on the backend side of things and that caused them to return a HTML page in the response. If we checked the response for an error we could update the appState with an error message saying that something went wrong on backend. Instead of it crashing and burning because we try to turn the HTML error page into a JSON object.

Reply:
I agree. At the least i feel that wrapping these calls in a try / catch would be best. If we do get an error the catch could return the the same type that the success would return. For example, if on success we return an array of objects, on error we'd return an empty array.
This would prevent the app from crashing, and let us know that something went wrong so we could then present a generic error message.

Reply:
Good point. I feel like we shouldn't be returning empty data back from these and would likely make more sense to throw an error and then have the consumer wrap these methods in a try/catch to handle errors?

Reply:
After looking at this more, I should also be wrapping the fetch request in a try catch and return an error before doing `response.json()` so we don't get the generic invalid token error that Jonny was mentioning.